src/parsers/enum_destructuring.cr

```diff
@@ -11,7 +12,17 @@ module Mint
 
         option = type_id! EnumDestructuringExpectedOption
 
-        parameters = many { type_variable }.compact
+        parameters = [] of Ast::TypeVariable
+
+        if char! '('
+          parameters.concat list(
+            terminator: ')',
+            separator: ','
+          ) { type_variable }.compact
+
+          whitespace
+          char ')', EnumDestructuringExpectedClosingParentheses
+        end
```

Comment:
Copied completely from the way Mint parses Enum initialization https://github.com/mint-lang/mint/blob/2659e3c793cbc8821da8c54258a80d575333efee/src/parsers/enum.cr#L18-L30